packages/generator-helper/src/dmmf.ts

```diff
@@ -1,56 +1,60 @@
+export type ReadonlyDeep<O> = {
+  +readonly [K in keyof O]: ReadonlyDeep<O[K]>
+}
+
 // eslint-disable-next-line @typescript-eslint/no-namespace
 export namespace DMMF {
-  export interface Document {
+  export type Document = ReadonlyDeep<{
```

Comment:
Did we got rid of last parts where we modify DMMF?

Reply:
I got rid of the ones I found, there were very few.

Reply:
Great! Should `ReadonlyDeep` also turn arrays in `ReadonlyArray`?

Reply:
I think it does, see eg.
![image](https://github.com/prisma/prisma/assets/18401805/b03160f5-a272-4982-a887-ed2f62846dc0)